src/phongo_apm.c

```diff
@@ -262,6 +262,36 @@ static void phongo_apm_command_failed(const mongoc_apm_command_failed_t* event)
 	FREE_HASHTABLE(subscribers);
 }
 
+static void phongo_apm_topology_changed(const mongoc_apm_topology_changed_t* event)
+{
+	mongoc_client_t*                   client;
+	HashTable*                         subscribers;
+	php_phongo_topologychangedevent_t* p_event;
+	zval                               z_event;
+
+	client      = mongoc_apm_topology_changed_get_context(event);
+	subscribers = phongo_apm_get_subscribers_to_notify(php_phongo_sdamsubscriber_ce, client);
+
+	/* Return early if there are no APM subscribers to notify */
+	if (zend_hash_num_elements(subscribers) == 0) {
+		goto cleanup;
+	}
+
+	object_init_ex(&z_event, php_phongo_topologychangedevent_ce); // todo
+	p_event = Z_TOPOLOGYCHANGEDEVENT_OBJ_P(&z_event);             // todo
```

Comment:
What are the `// todo` comments referring to?

Reply:
Sorry, forgot to clean that up!